Sources/MapboxDirections/MapMatching/MapMatchingResponse.swift

```diff
@@ -36,7 +36,8 @@ class MapMatchingResponse: Decodable {
             // Postprocess each route.
             for route in routes {
                 // Imbue each route’s legs with the leg-separating waypoints refined above.
-                route.legSeparators = waypoints.filter { $0.separatesLegs }
+                // TODO: Filter these waypoints by whether they separate legs, based on the options, if given.
```

Comment:
@1ec5 , so here we should filter out `waypoints` so that we leave only ones corresponding `options.legSeparators` (if options are present)? Is it correct?